lib/sth_database.js

```diff
@@ -1414,25 +1586,257 @@ function storeRawData(data, callback) {
+    ).toArray(
+      function (err, restArray) {
+        if (callback) {
+          return process.nextTick(
+            callback.bind(null, err, (restArray && restArray.length) ? restArray[0].attrValue : null));
+        }
+      }
+    );
+  }
+  var getNewMaxValue4Resolutions = {};
+  for (var i = 0; i < sthConfig.AGGREGATION_BY.length; i++) {
+    getNewMaxValue4Resolutions[sthConfig.AGGREGATION_BY[i]] =
+      getNewMaxValue.bind(null, data, sthConfig.AGGREGATION_BY[i]);
+  }
+  async.parallel(
+    getNewMaxValue4Resolutions,
+    function (err, result) {
       if (callback) {
-        callback(err);
+        return process.nextTick(callback.bind(null, err, result));
       }
-    });
+    }
+  );
+}
+
+/**
+ * Asynchronously returns the new minimum values for the resolutions of interest after some raw data update
+ * @param {object} data The data to be stored. It is an object including the following properties:
+ *  - {string} entityId The entity id associated to the updated attribute
+ *  - {string} entityType The entity type associated to the updated attribute
+ *  - {object} attribute The updated attribute
+ *  - {Date} timestamp The timestamp associated to the updated raw data
+ * @param callback The callback to notify in case of error or once the new maximum value has been calculated
+ */
+function getNewMinValues(data, callback) {
+  function getNewMinValue(data, resolution, callback) {
+    data.collection.find(
+      getNewMinMaxCondition(data, resolution)
+    ).sort(
+      {
+        attrValue: -1
+      }
+    ).toArray(
+      function (err, restArray) {
+        if (callback) {
+          return process.nextTick(
+            callback.bind(null, err, (restArray && restArray.length) ? restArray[0].attrValue : null));
+        }
+      }
+    );
+  }
+  var getNewMinValue4Resolutions = {};
+  for (var i = 0; i < sthConfig.AGGREGATION_BY.length; i++) {
+    getNewMinValue4Resolutions[sthConfig.AGGREGATION_BY[i]] =
+      getNewMinValue.bind(null, data, sthConfig.AGGREGATION_BY[i]);
+  }
+  async.parallel(
+    getNewMinValue4Resolutions,
+    function (err, result) {
+      if (callback) {
+        return process.nextTick(callback.bind(null, err, result));
+      }
+    }
+  );
+}
+
+/**
+ * Analyzes the search of already existent raw data and invokes the passed callback with the results
+ *  (this is, the information about the notification)
+ * @param {object} data The data to be stored. It is an object including the following properties:
+ *  - {object} collection: The collection where the data should be stored in
+ *  - {date} recvTime The date the event arrived
+ *  - {string} entityId The entity id associated to the updated attribute
+ *  - {string} entityType The entity type associated to the updated attribute
+ *  - {object} attribute The updated attribute
+ * @param {Function} callback Function to call once the operation completes
+ */
+function onAlreadyRegisteredDataSearch(data, callback, err, result) {
```

Comment:
The Node convention stablish that error should be the first parameter and callback the last one. Though this is not mandatory, it could ease future refactorings a lot.